runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/jobsubmission/JobInvocation.java

```diff
@@ -70,8 +72,9 @@ public JobInvocation(
     this.stateObservers = new ArrayList<>();
     this.messageObservers = new ArrayList<>();
     this.invocationFuture = null;
-    this.jobState = JobState.Enum.STOPPED;
+    this.stateHistory = new ArrayList<>();
```

Comment:
Does this grow boundlessly?

Reply:
yes, but based on the feedback from @lukecwik in the last job-state PR, there are no runners that can switch back and forth between states, like STOPPED -> RUNNING -> STOPPED -> RUNNING, so the history should always be quite limited.